addons/im_livechat/controllers/main.py

```diff
@@ -103,13 +101,14 @@ def get_session(self, channel_id, previous_operator_id=None, chatbot_script_id=N
             chatbot_script_id=chatbot_script_id,
             country_id=country.id,
             lang=request.cookies.get("frontend_lang"),
-            **extra_operator_lookup_params
+            **kwargs
         )
         if not operator_info['operator_partner']:
             return False
 
         chatbot_script = operator_info['chatbot_script']
         is_chatbot_script = operator_info['operator_model'] == 'chatbot.script'
+        processed_extra_channel_params = self.env['discuss.channel']._process_extra_channel_params(kwargs)
```

Comment:
`_process_extra_channel_params` should belong to the controller as it's only called here and it doesn't need a channel record